File: android-room-sample-master/app/src/javaCommon/java/com/emreeran/android/roomsample/db/dao/UserDao.java
package com.emreeran.android.roomsample.db.dao;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import com.emreeran.android.roomsample.db.entity.User;

import java.util.List;

import io.reactivex.Single;

/**
 * Created by Emre Eran on 25.04.2018.
 */
@Dao
public interface UserDao {
    @Insert(onConflict = OnConflictStrategy.REPLACE)
    long insert(User user);

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    void insertAll(List<User> users);

    @Query("SELECT * FROM users")
    Single<List<User>> list();

    @Query("DELETE FROM users")
    void purge();
}

File: android-room-sample-master/app/src/kotlinCommon/java/com/emreeran/android/roomsample/db/dao/UserDao.kt
package com.emreeran.android.roomsample.db.dao

import android.arch.persistence.room.Dao
import android.arch.persistence.room.Insert
import android.arch.persistence.room.OnConflictStrategy
import android.arch.persistence.room.Query
import com.emreeran.android.roomsample.db.entity.User
import io.reactivex.Single

/**
 *  Created by Emre Eran on 20.04.2018.
 */
@Dao
interface UserDao {
    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insert(user: User)

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAll(users: List<User>)

    @Query("SELECT * FROM users")
    fun list(): Single<List<User>>

    @Query("DELETE FROM users")
    fun purge()
}
